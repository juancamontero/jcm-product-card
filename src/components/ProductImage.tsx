import React, { useContext } from 'react';
import { ProductContext } from './ProductCard';

import styles from '../styles/styles.module.css';
import noImage from '../assets/no-image.jpg';

export interface Props {
  img?: string;
  className?: string;
  style?: React.CSSProperties;
}

export const ProductImage = ({ img, className, style }: Props) => {
  const imgSrc = img ? img : useContext(ProductContext).product.img;
  const altText = !imgSrc
    ? 'no image icon'
    : 'Product image'

  return (
    <img
      className={` ${styles.productImg} ${className}`}
      src={imgSrc ? imgSrc : noImage}
      alt={altText}
      style={style}
    />
  );
};
