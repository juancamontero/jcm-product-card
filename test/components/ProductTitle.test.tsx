import * as React from 'react';
import renderer from 'react-test-renderer';
import { ProductTitle, ProductCard } from '../../src/components';
import { product1 } from '../data/products';

describe('ProductTitle Tests', () => {
  test('should render the component correctly', () => {
    const wrapper = renderer.create(<ProductTitle title="custom title" />);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('should shows the component with the PRODUCT title', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>{() => <ProductTitle />}</ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
