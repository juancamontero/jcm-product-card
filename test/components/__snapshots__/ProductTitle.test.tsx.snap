// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle Tests should render the component correctly 1`] = `
<span
  className=" productDescription undefined"
>
  custom title
</span>
`;

exports[`ProductTitle Tests should shows the component with the PRODUCT title 1`] = `
<div
  className="productCard undefined"
>
  <span
    className=" productDescription undefined"
  >
    Coffee Mug - with image
  </span>
</div>
`;
